### PR TITLE
Remove Windows arm32 support

### PR DESCRIPTION
**Pull request by BruceForstall**
Mostly, remove the CI runs. Windows arm32 product support was removed before .NET 6.

### CODE REVIEW

**Comment by BruceForstall** [comment]
cc @dotnet/jit-contrib

**Comment by BruceForstall** [comment]
Comments, anyone?

There are probably many more things that could be removed after this is merged. The primary goal here was to stop running win-arm builds and tests in the CI system.

File: eng/pipelines/coreclr/templates/build-job.yml [hunk 1]
```diff
@@ -202,10 +202,6 @@ jobs:
       - script: $(Build.SourcesDirectory)/src/coreclr/build-runtime$(scriptExt) $(buildConfig) $(archType) -hostarch x64 $(osArg) -ci $(compilerArg) -component crosscomponents -cmakeargs "-DCLR_CROSS_COMPONENTS_BUILD=1" $(officialBuildIdArg) $(clrRuntimePortableBuildArg)
         displayName: Build CoreCLR Cross-Arch Tools (Tools that run on x64 targeting x86)
 
-    - ${{ if and(eq(parameters.osGroup, 'windows'), eq(parameters.archType, 'arm')) }}:
-      - script: $(Build.SourcesDirectory)/src/coreclr/build-runtime$(scriptExt) $(buildConfig) $(archType) -hostarch x86 $(osArg) -ci $(compilerArg) -component crosscomponents -cmakeargs "-DCLR_CROSS_COMPONENTS_BUILD=1" $(officialBuildIdArg) $(clrRuntimePortableBuildArg)
-        displayName: Build CoreCLR Cross-Arch Tools (Tools that run on x86 targeting arm)
-
```
Comment (jkoritzinsky) [inline eng/pipelines/coreclr/templates/build-job.yml:208]: There's a condition in `eng/Subsets.props` that corresponds to this condition. Do we want to remove that as well?

**Review by jkoritzinsky — APPROVED** [review]
LGTM. Do we want to remove support for building the product build for Windows ARM32, or just the runtime test build/run?

**Comment by jkotas** [comment]
There is more that can be cleaned up in the sources (e.g. delete src\coreclr\vm\arm\*.asm). It can be done as a separate change.

**Review by jkotas — APPROVED** [review]

File: src/tests/run.cmd [hunk 2]
```diff
@@ -200,7 +199,7 @@ echo.
 echo where:
 echo.
 echo./? -? /h -h /help -help   - View this message.
-echo ^<build_architecture^>      - Specifies build architecture: x64, x86, arm, or arm64 ^(default: x64^).
+echo ^<build_architecture^>      - Specifies build architecture: x64, x86, or arm64 ^(default: x64^).
```
Comment (trylek) [inline src/tests/run.cmd:202]: Hmm, I'm still seeing linux_arm / linux_musl_arm jobs in the above pipelines, are we removing arm32 support even on these?

File: src/tests/issues.targets [hunk 3]
```diff
@@ -291,235 +291,6 @@
...
-        </ExcludeList>
-        <ExcludeList Include="$(XunitTestBinBase)/profiler/elt/slowpatheltleave/**">
-            <Issue>https://github.com/dotnet/runtime/issues/84750</Issue>
-        </ExcludeList>
-        <ExcludeList Include="$(XunitTestBinBase)/tracing/eventpipe/eventsourceerror/eventsourceerror/*">
-            <Issue>https://github.com/dotnet/runtime/issues/81241</Issue>
-        </ExcludeList>
-    </ItemGroup>
-
     <!-- Windows arm64 specific excludes -->
```
Comment (trylek) [inline src/tests/issues.targets:294]: Not as part of this PR but I guess at some point we should audit this issue list and close those that are no longer relevant i.e. that only occur on the no longer supported Windows arm platform.

**Review by trylek — APPROVED** [review]
LGTM overall modulo one nitpick. I guess we'll likely want to make an equivalent change in the .NET 6 / 7 servicing branches, correct?

File: src/tests/run.cmd (hunk 2 above)
Comment (BruceForstall) [reply src/tests/run.cmd:202]: No, linux-arm is still a supported platform. (But run.cmd is only used on Windows)

**Comment by BruceForstall** [comment]
> I guess we'll likely want to make an equivalent change in the .NET 6 / 7 servicing branches, correct?

Since we don't support win-arm in .NET 6/7, that would make sense.

File: eng/pipelines/coreclr/templates/build-job.yml (hunk 1 above)
Comment (BruceForstall) [reply eng/pipelines/coreclr/templates/build-job.yml:208]: Yes. But that will be a follow-up.

**Comment by BruceForstall** [comment]
I decided to simplify this to only remove win-arm CI runs. Removing the ability to build via build scripting can happen later.

**Pull request merged**